src/debugging-support-in-rustc.md

```diff
@@ -0,0 +1,318 @@
+
+Main issue in achieving this is setting up dependencies. One such dependency is Python. That
+is why we have our own fork of GDB because one of the drivers is patched on Rust's side to
+check the correct version of Python (Python 2.7 in this case. *Note: Python3 is not chosen
+for this purpose because Python's stable ABI is limited and is not sufficient for GDB's needs.
+See [https://docs.python.org/3/c-api/stable.html]*).
+
+This is to keep updates to debugger as fast as we make changes to the debugging symbols.
+In essence, to ship the debugger as soon as new debugging info is added. GDB only releases
+every six months or so and is comparably slower release cycle. However, the changes that are
+not related to Rust itself should ideally be first merged to upstream eventually.
+
+#### Code signing for LLDB debug server on macOS
+
+According to Wikipedia, [System Integrity Protection] is
+
+> System Integrity Protection (SIP, sometimes referred to as rootless) is a security feature
+> of Apple's macOS operating system introduced in OS X El Capitan. It comprises a number of
+> mechanisms that are enforced by the kernel. A centerpiece is the protection of system-owned
+> files and directories against modifications by processes without a specific "entitlement",
+> even when executed by the root user or a user with root privileges (sudo).
+
+It prevents processes using `ptrace` syscall. If a process wants to use `ptrace` it has to be
+code signed. The certificate that signs it has to be trusted on your machine.
+
+See [Apple developer documentation for System Integrity Protection].
+
+We may need to sign up with Apple and get the keys to do this signing. Tom has looked into if
+Mozilla can do this but Mozilla cannot because Mozilla is at the maximum number of things that
+keys Mozilla has are allowed to sign. Tom does not know if Mozilla could get more keys.
+
+Alternatively, Tom suggests that maybe a Rust legal entity is needed to get the keys via Apple.
+This problem is not technical in nature. If we had such a key we could sign GDB as well and
+ship that.
+
+#### DWARF and Traits
+
+Rust traits are not emitted into DWARF at all. The impact of this is calling a method `x.method()`
+does not work as is. The reason being that method is implemented by a trait, as opposed
+to a type. That information is not present so finding trait methods is missing.
+
+DWARF has a notion of interface types (possibly added for Java). Tom's idea was to use this
+interface type as traits.
+
+DWARF only deals with concrete names, not the reference types. So, a given implementation of a
+trait for a type would be one of these interfaces (`DW_tag_interface` type). Also, the type for
+which it is implemented would describe all the interfaces this type implements. This requires a
+DWARF extension.
+
+Issue on Github: [https://github.com/rust-lang/rust/issues/33014]
+
+##### Typical process for a Debug Info change (LLVM)
```

Comment:
```suggestion
## Typical process for a Debug Info change (LLVM)
```

Reply:
Could you move this too a new section after the `Procedural macro stepping` section?